src/client/service-components/TextToAdd.js

```diff
@@ -0,0 +1,94 @@
+import React, {useContext, useRef, useState} from 'react';
+import {UserSettingsContext} from '../context-store/user-settings-context.js';
+import {showErrorMessage} from '../utils/notifier.js';
+import {ToastContainer, Slide} from 'react-toastify';
+import useOnClickOutside from 'use-onclickoutside';
+
+import './TextToAdd.css';
+import 'react-toastify/dist/ReactToastify.css';
+
+
+export const TextToAdd = ({saveItem, itemTexts, showError = false, type}) => {
+    const {language, theme} = useContext(UserSettingsContext);
+
+    const [text, setText] = useState('');
+    const [isAdding, setIsAdding] = useState(false);
+    const closingRef = useRef();
+    useOnClickOutside(closingRef, () => setIsAdding(false));
+
+    const updateText = ({target}) => {
+        setText(target.value);
+    };
+
+    const keyPressedHandle = async ({key}) => {
```

Comment:
Why is it async?

Reply:
It left from times I had server-requests there :) Fixed. Commit 3afd4b9c2053634f1d6587eba6018cb3f1e738ee